wandb/sdk/launch/agent/agent.py

```diff
@@ -305,7 +299,18 @@ def finish_thread_id(
             except CommError:
                 pass
             if run_info is None:
```

Comment:
do we maybe also want to have a simple/trivial case where we check the health of `run_info`? Im not sure if this is relevant, but what if we get info back that doesn't look right? Like the run is in the wrong state? We only ever use this fail _run_queue_item mutation when we get **nothing** out of the api call here, which is sort of weird. Maybe a comment that mentioned we never ever use the output of this call? or make a helper function so you can do something like: 
```python
try:
    self._api.get_run_info(
             self._entity, job_and_run_status.project, job_and_run_status.run_id
    )
    # success! 
    return
except:
    pass

# do our error handling here (or in the except)
```

Reply:
I'll add a TODO and file a ticket